drivers/storage/portworx/component/windows.go

```diff
@@ -0,0 +1,163 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+const (
+	WindowsComponentName     = "Windows Node"
+	WindowsDaemonSetName     = "csi-pwx-node-win"
+	WindowsStorageClass      = "px-csi-win"
+	WindowsDaemonSetFileName = "win.yaml"
+)
+
+type windows struct {
+	client        client.Client
+	isCreated     bool
+	isWindowsNode bool
+}
+
+func (w *windows) Initialize(
+	k8sClient client.Client,
+	_ version.Version,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+) {
+	w.client = k8sClient
+}
+
+func (w *windows) Name() string {
+	return WindowsComponentName
+}
+
+func (w *windows) Priority() int32 {
+	return DefaultComponentPriority
+}
+
+func (w *windows) IsPausedForMigration(cluster *corev1.StorageCluster) bool {
+	return util.ComponentsPausedForMigration(cluster)
+}
+
+func (w *windows) IsEnabled(cluster *corev1.StorageCluster) bool {
+	w.isWindowsNode = true
+	return w.isWindowsNode
+}
+
+func (w *windows) Reconcile(cluster *corev1.StorageCluster) error {
+	ownRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	var errList []string
+
+	if err := w.createDaemonSet(WindowsDaemonSetFileName, ownRef, cluster); err != nil {
+		logrus.Errorf("error during creating %s daemonset %s ", WindowsDaemonSetName, err)
+		errList = append(errList, err.Error())
+	}
+
+	if err := w.createStorageClass(); err != nil {
+		logrus.Errorf("error during creating %s storageclass %s ", WindowsStorageClass, err)
+		errList = append(errList, err.Error())
+	}
+
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (w *windows) Delete(cluster *corev1.StorageCluster) error {
+
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := k8s.DeleteDaemonSet(w.client, WindowsDaemonSetName, cluster.Namespace, *ownerRef); err != nil {
+		return err
+	}
+	w.MarkDeleted()
+	return nil
+}
+
+func (w *windows) MarkDeleted() {
+	w.isCreated = false
+}
+
+// RegisterPortworxPluginComponent registers the PortworxPlugin component
+func RegisterWindowsComponent() {
+	Register(WindowsComponentName, &windows{})
+}
+
+func init() {
+	RegisterWindowsComponent()
+}
+
+func (w *windows) createDaemonSet(filename string, ownerRef *metav1.OwnerReference, cluster *corev1.StorageCluster) error {
+	daemonSet, err := k8s.GetDaemonSetFromFile(filename, pxutil.SpecsBaseDir())
+	if err != nil {
+		return err
+	}
+
+	daemonSet.Namespace = cluster.Namespace
+	// daemonSet.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+
+	existingDaemonSet := &appsv1.DaemonSet{}
+	getErr := w.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      WindowsDaemonSetName,
+			Namespace: cluster.Namespace,
+		},
+		existingDaemonSet,
+	)
+	if getErr != nil && !errors.IsNotFound(getErr) {
+		return getErr
+	}
+
+	// pxutil.ApplyStorageClusterSettingsToPodSpec(cluster, &daemonSet.Spec.Template.Spec)
```

Comment:
We need to add this, otherwise the DS will get created on nodes where pod is not supposed to be deployed or it won't honor the image registry, pull policy, etc from the stroageclass.

Reply:
The only reason this is commented right now is because this is overwrting nodeselector from storagecluster, for windows support we have nodeaffinity as os=linux and for this daemonset, we need nodeaffinity as os=windows.

This is planned to be taken in next PR.